cfgov/unprocessed/js/organisms/FilterableListControls.js

```diff
@@ -18,157 +17,254 @@ var Notification = require( '../molecules/Notification' );
 
-    var field;
-    var name;
-    for ( var f = 0, flen = fields.length; f < flen; f++ ) {
-      field = fields[f];
-      name = field.getAttribute( 'name' );
-      if ( !checkgroups[name] ) {
-        checkgroups[name] = [];
+      for ( var prop in validatedField.status ) {
+        if ( validatedField.status[prop] === false ) fieldIsValid = true;
       }
-      checkgroups[name].push( field );
-    }
 
-    var status;
-    for ( var group in checkgroups ) {
-      if ( checkgroups.hasOwnProperty( group ) ) {
-        status = group.length > 1 ?
-                 _validateCheckGroup( group ) : _validateInput( group );
-
-        for ( var prop in status.status ) {
-          if ( status.status[prop] === false ) {
-            validatedFields.valid.push( field );
-          } else {
-            validatedFields.invalid.push( field );
-          }
-        }
+      if ( fieldIsValid ) {
+        validatedFields.invalid.push( validatedField );
+      } else {
+        validatedFields.valid.push( validatedField );
       }
-    }
+    } );
 
     return validatedFields;
   }
 
   /**
-   * @param {Object} elem The check group we're testing.
-   * @returns {Object}
-   *   The formatted validation object of the tested check group.
+   * Validate the specific field types.
+   * @param {Field} field A form field.
+   * @param {Boolean} isInGroup A boolean that determines if field in a group.
+   * @returns {Object} An object with a status and message properties.
    */
-  function _validateCheckGroup( elem ) {
-    // TODO: Replace mock data with real validation data.
-    return {
-      elem:   elem,
-      value:  null,
-      label:  null,
-      status: {
-        checkgroup: false
-      }
+  function _validateField( field, isInGroup ) {
+    var emptyStringRegex = /^\s*$/;
+    var value = field.value;
+    var isRequired = field.getAttribute( 'required' ) !== null;
+    var isEmpty = value === '' || value === null ||
+                  typeof value === 'undefined' ||
+                  emptyStringRegex.test( value );
+    var type = field.getAttribute( 'data-type' ) ||
+               field.getAttribute( 'type' ) ||
+               field.tagName.toLowerCase();
+    var validation = {
+      field:      field,
+      // TODO: Change layout of field groups to use fieldset.
+      label:      _getLabelText( field, '', false || isInGroup ),
+      msg:        '',
+      status:     null
     };
-  }
 
-  /**
-   * @param {Object} elem The input we're testing.
-   * @returns {Object} The formatted validation object of the tested input.
-   */
-  function _validateInput( elem ) {
-    // TODO: Replace mock data with real validation data.
-    return {
-      elem:   elem,
-      value:  null,
-      label:  null,
-      status: false
-    };
+
+    if ( _validators[type] ) {
+      validation.status = _validators[type]( field, validation );
+    }
+
+    if ( isRequired && isEmpty ) {
+      validation.msg += 'is required.';
```

Comment:
fixed in https://github.com/cfpb/cfgov-refresh/commit/f217937d4eec3537f28765b2f956d29b1609df50